registry/proxy/proxyauth.go

```diff
@@ -19,7 +19,7 @@ type credentials struct {
 }
 
 func (c credentials) Basic(u *url.URL) (string, string) {
-	up := c.creds[u.String()]
+	up := c.creds[tokenURL]
```

Comment:
`up := c.creds[u.String()]` is correct, this particular basic auth should only be added when contacting the tokenURL. This code would cause the credentials to always be added even to endpoints to which they do not belong, which has the effect of sharing username/password with an unintended audience. Line 34 below enforces that the username and password are only used when contacting `tokenURL`.  To implement non-hub support `ConfigureAuth` needs to be updated to use the realm that is returned from the ping, not `remoteURL` (the token server could be elsewhere). `ping` should also be updated to return the realm. If basic and not token auth is used (ping would determine this), then it is appropriate to use `remoteURL`.